okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/Platform.java

```diff
@@ -105,47 +119,71 @@ public void connectSocket(Socket socket, InetSocketAddress address,
   }
 
   /** Attempt to match the host runtime to a capable Platform implementation. */
+  @SuppressWarnings("unchecked")
   private static Platform findPlatform() {
-    // Attempt to find Android 2.3+ APIs.
+    // Find the conscrypt security provider if we can
+    Class<Provider> conscryptProviderClass = null;
     try {
+      conscryptProviderClass =
+          (Class<Provider>) Class.forName("org.conscrypt.OpenSSLProvider");
+    } catch (Throwable t) {
       try {
-        Class.forName("com.android.org.conscrypt.OpenSSLSocketImpl");
-      } catch (ClassNotFoundException e) {
-        // Older platform before being unbundled.
-        Class.forName("org.apache.harmony.xnet.provider.jsse.OpenSSLSocketImpl");
+        conscryptProviderClass =
+            (Class<Provider>)  Class.forName("com.android.org.conscrypt.OpenSSLProvider");
+      } catch (Throwable t2) {
+        try {
+          conscryptProviderClass = (Class<Provider>)
+              Class.forName("org.apache.harmony.xnet.provider.jsse.OpenSSLProvider");
+        } catch (Throwable t3) {
+          // Stick with what we have
+        }
+      }
+    }
+    boolean haveConscrypt = false;
+    Provider sslProvider = null;
+    try {
+      sslProvider = SSLContext.getDefault().getProvider();
+    } catch (NoSuchAlgorithmException nsae) {
+      // Ignore
+    }
+    if (conscryptProviderClass != null) {
+      if (sslProvider != null && sslProvider.getClass().equals(conscryptProviderClass)) {
+        haveConscrypt = true;
+      } else {
+        try {
+          sslProvider = conscryptProviderClass.newInstance();
```

Comment:
Yes, that will be the case on AppEngine which cannot switch the default without a non-trivial likelihood of breaking people. Future releases might change that